android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java

```diff
@@ -1461,6 +1461,10 @@ protected void registerForTouch(final View touchable)
 		boolean clickable = true;
 		if (proxy.hasProperty(TiC.PROPERTY_TOUCH_ENABLED)) {
 			clickable = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_TOUCH_ENABLED), true);
+			//If touchEnabled is specifically set, enable or disable touch sound
+			touchable.setSoundEffectsEnabled(clickable);
+		} else {
+			touchable.setSoundEffectsEnabled(false);
```

Comment:
I'm not sure. The default behavior of TI is that touch sounds are enabled on every kind of view, no matter whether or not that view has any kind of touch / click event applied to it. This not the default behavior in native-built applications.  If you don't do this is some kind of global fashion, then you would have to apply this setting to every nested element of a view.  Say a view in comprised of three additional views, all without interactivity, then you would specifically have to apply touchEnabled = false to each and every view. To make matters worse, on iOS setting touchEnabled = false on a parent view, would disable interactive events on all child views (not true in Android). As such, this was the only feasible place I could think / find to implement this logic.

The only way a developer has to make this choice currently is styles.xml, which is an all or nothing proposition.

<style name="AppTheme" parent="AppBaseTheme">
    <item name="android:soundEffectsEnabled">false</item>
</style>